### PR TITLE
8326333: jshell <TAB> completion on arrays is incomplete

### PR DESCRIPTION
**Pull request by lahodaj**
Inside jshell, arrays don't show `Object` method in array code completion, i.e.:
```
jshell> String[] arr = null;
arr ==> null

jshell> arr.<code-completion-here>
```

will only show `length`. This patch adds `j.l.Object` methods to the suggestions for arrays.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326333](https://bugs.openjdk.org/browse/JDK-8326333): jshell &lt;TAB&gt; completion on arrays is incomplete (**Bug** - P4)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18163/head:pull/18163` \
`$ git checkout pull/18163`

Update a local copy of the PR: \
`$ git checkout pull/18163` \
`$ git pull https://git.openjdk.org/jdk.git pull/18163/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18163`

View PR using the GUI difftool: \
`$ git pr show -t 18163`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18163.diff">https://git.openjdk.org/jdk/pull/18163.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18163#issuecomment-1985284190)

### CODE REVIEW

File: test/langtools/jdk/jshell/CompletionSuggestionTest.java [hunk 1]
```diff
@@ -810,4 +810,15 @@ public void testDuplicateImport() {
                                      ste(m2, RECOVERABLE_DEFINED, VALID, true, MAIN_SNIPPET));
         assertSignature("test(|", "void test(String s)", "void test(Integer i)");
     }
+
+    //JDK-8326333
```
Comment (lgxbslgx) [inline test/langtools/jdk/jshell/CompletionSuggestionTest.java:814]: Such simple bug-id seems unnecessary or uncomplete.  It is better to privode some description content.

**Review by lgxbslgx — CHANGES_REQUESTED** [review]
The copyright of `SourceCodeAnalysisImpl.java` needs to be adjusted too. Thanks.

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Pull request closed without merging**